Scripts/Runtime/Entities/TaskDriver/TaskSet/Job/Wrapper/AbstractDataStreamPendingAccessWrapper.cs

```diff
@@ -0,0 +1,79 @@
+using Anvil.CSharp.Logging;
+using Anvil.Unity.DOTS.Jobs;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Anvil.Unity.DOTS.Entities.TaskDriver
+{
+    internal abstract class AbstractDataStreamPendingAccessWrapper<T> : AbstractAccessWrapper
+        where T : AbstractDataStream, IAbstractDataStream
+    {
+        protected readonly T m_DefaultStream;
+
+        protected AbstractDataStreamPendingAccessWrapper(T defaultStream, AccessType accessType, AbstractJobConfig.Usage usage)
+            : base(accessType, usage)
+        {
+            m_DefaultStream = defaultStream;
+            DEBUG_TrackRequiredStream(defaultStream);
+        }
+
+        public T GetInstance(IAbstractDataStream explicitStream = null)
+        {
+            DEBUG_EnforceExplicitStream(explicitStream);
+            DEBUG_EnsureStreamWasRequired(explicitStream);
+
+            return (T)explicitStream ?? m_DefaultStream;
+        }
+
+        public override void MergeStateFrom(AbstractAccessWrapper other)
+        {
+            base.MergeStateFrom(other);
+
+            AbstractDataStreamPendingAccessWrapper<T> otherTyped = (AbstractDataStreamPendingAccessWrapper<T>)other;
+            m_DEBUG_RequiredStreams.UnionWith(otherTyped.m_DEBUG_RequiredStreams);
+            DEBUG_TrackRequiredStream(otherTyped.m_DefaultStream);
+        }
+
+        //*************************************************************************************************************
+        // SAFETY
+        //*************************************************************************************************************
+
+        // #if ANVIL_DEBUG_SAFETY
+        private HashSet<T> m_DEBUG_RequiredStreams;
+
+        [Conditional("ANVIL_DEBUG_SAFETY")]
+        public void DEBUG_TrackRequiredStream(T stream)
```

Comment:
Should this be private?

Reply:
Good catch, This was originally called from the outside but the last minute addition of `MergeStateFrom` brought this internal.